File: benches/py/sieve.py
import sys
n = int(sys.argv[1])
is_prime = [True] * (n + 1)
if n >= 0: is_prime[0] = False
if n >= 1: is_prime[1] = False
count = 0
for p in range(2, n + 1):
    if is_prime[p]:
        count += 1
        step = p
        j = p * p
        while j <= n:
            is_prime[j] = False
            j += step
print(count)

File: benches/lua/sieve.lua
local n = tonumber(arg[1])
local is_prime = {}
for i = 0, n do is_prime[i] = true end
if n >= 0 then is_prime[0] = false end
if n >= 1 then is_prime[1] = false end
local count = 0
for p = 2, n do
    if is_prime[p] then
        count = count + 1
        local step = p
        local j = p * p
        while j <= n do
            is_prime[j] = false
            j = j + step
        end
    end
end
print(count)
